File: 24.07.2024/Stack.py
class Stack:
    def __init__(self):
        self.stack = []
    def push(self, x):
        self.stack.append(x)
    def pop(self):
        if not self.is_empty():
            return self.stack.pop()
        return "Stack Underflow"
    def peek(self):
        if not self.is_empty():
            return self.stack[-1]
        return "Stack is Empty"

    def is_empty(self):
        return len(self.stack) == 0

s = Stack()
s.push(10)
s.push(20)
s.push(30)
print(s.pop(), "popped from stack")
print("Top element is", s.peek())

File: 24.07.2024/stack.c
#include <stdio.h>
#define MAX 1000
int stack[MAX];
int top = -1;
void push(int x) {
    if (top >= MAX - 1) {
        printf("Stack Overflow\n");
        return;
    }
    stack[++top] = x;
}
int pop() {
    if (top < 0) {
        printf("Stack Underflow\n");
        return -1;
    }
    return stack[top--];
}
int peek() {
    if (top < 0) {
        printf("Stack is Empty\n");
        return -1;
    }
    return stack[top];
}
int isEmpty() {
    return top < 0;
}
int main() {
    push(10);
    push(20);
    push(30);
    printf("%d popped from stack\n", pop());
    printf("Top element is %d\n", peek());
    return 0;
}
